### DISCUSSION TITLE
How to pass parameters to middleware

### DISCUSSION THREAD

**Discussion by KamilSawicki** [spatie/laravel-route-attributes#104] [Q&A]
Hi. How can I use middleware with parameters? For example, I have middleware handle(Request $request, Closure $next, string $role) and I cannot find how to pass role parameter value.

**Comment by msacharewicz** [comment]
You can use the same notation as with the standard `middleware()` method, as nicely shown [here](https://laravel.com/docs/9.x/authorization#via-middleware) and [here](https://laravel.com/docs/9.x/routing#attaching-rate-limiters-to-routes).

Basically, use a `"yourmiddleware:arguments"` format:

```php
#[Patch('your-path/{yourModel}', middleware: "yourmiddleware:argument1,argument2,yourModel,...")]
```
so, in your case, simply:

```php
#[Patch('your-path/{yourModel}', middleware: "yourmiddleware:MyFancyRole")]
```
assuming that you want to pass a `"MyFancyRole"` string to your `$role` attribute.

The `yourmiddleware` part is the key you have provided in `/app/Http/Kernel.php` middleware registration:

```php
protected $routeMiddleware = [
    // ...
    'yourmiddleware' => \App\Http\Middleware\YourMiddleware::class,
    // ...
];
```
---

Currently, the documentation is a bit vague on it, providing only the `middleware: SomeMiddleware::class` notation in usage examples.

It wouldn't hurt, if the `README.md` provided some commonly-used example, like with gates/authorization:

```php
#[Put('/post/{post}', middleware: "can:update,post")]
```
as an equivalent of Laravel's example:
```php
use App\Models\Post;
 
Route::put('/post/{post}', function (Post $post) {
    // The current user may update the post...
})->middleware('can:update,post');
```
